onnxruntime/core/graph/graph.cc

```diff
@@ -467,6 +480,188 @@ void Node::ToProto(NodeProto& proto, bool update_subgraphs) const {
+  nb.add_index(gsl::narrow<uint32_t>(index_));
+  nb.add_op_type(op_type);
+  nb.add_type(static_cast<fbs::NodeType>(node_type_));
+  nb.add_execution_provider_type(ep);
+  nb.add_inputs(inputs);
+  nb.add_outputs(outputs);
+  nb.add_attributes(attributes);
+  nb.add_input_arg_counts(input_arg_counts);
+  nb.add_implicit_inputs(implicit_inputs);
+  fbs_node = nb.Finish();
+  return Status::OK();
+}
+
+flatbuffers::Offset<fbs::NodeEdge> Node::SaveEdgesOrtFormat(flatbuffers::FlatBufferBuilder& builder) const {
+  auto get_edges = [](const EdgeSet& edge_set) {
+    std::vector<fbs::EdgeEnd> edges;
+    edges.reserve(edge_set.size());
+    for (const auto& edge : edge_set)
+      edges.push_back(fbs::EdgeEnd(gsl::narrow<uint32_t>(edge.GetNode().Index()),
+                                   edge.GetSrcArgIndex(), edge.GetDstArgIndex()));
+
+    return edges;
+  };
+
+  const auto input_edges = get_edges(relationships_.input_edges);
+  const auto output_edges = get_edges(relationships_.output_edges);
+  return fbs::CreateNodeEdgeDirect(builder, gsl::narrow<uint32_t>(index_), &input_edges, &output_edges);
+}
+
+#endif  // !defined(ORT_MINIMAL_BUILD)
+
+Status Node::LoadFromOrtFormat(const onnxruntime::experimental::fbs::Node& fbs_node, Graph& graph,
+                               const logging::Logger& logger, std::unique_ptr<Node>& node) {
+  node.reset(new Node(fbs_node.index(), graph));
+  return node->LoadFromOrtFormat(fbs_node, logger);
+}
+
+#define FBS_SET_STR_VAL(VAR, FBS_STR) \
+  if (FBS_STR) {                      \
+    VAR = FBS_STR->str();             \
+  }
+
+Status Node::LoadFromOrtFormat(const onnxruntime::experimental::fbs::Node& fbs_node, const logging::Logger& logger) {
+  auto LoadNodeArgsOrtFormat =
+      [&](const flatbuffers::Vector<flatbuffers::Offset<flatbuffers::String>>* fbs_node_arg_names,
+          std::vector<NodeArg*>& node_args,
+          bool check_parent_graph = false) -> Status {
+    ORT_RETURN_IF(nullptr == fbs_node_arg_names, "fbs_node_arg_names cannot be null");
+    node_args.reserve(fbs_node_arg_names->size());
+    if (fbs_node_arg_names) {
+      for (const auto& node_arg_name : *fbs_node_arg_names) {
+        ORT_RETURN_IF(nullptr == node_arg_name, "node_arg_name cannot be null");
+        auto* node_arg = check_parent_graph ? graph_->GetNodeArgIncludingParentGraphs(node_arg_name->str())
+                                            : graph_->GetNodeArg(node_arg_name->str());
+        ORT_RETURN_IF(nullptr == node_arg, "Could not find NodeArg ", node_arg_name->str());
+        node_args.push_back(node_arg);
+      }
+    }
+    return Status::OK();
+  };
+
+  FBS_SET_STR_VAL(name_, fbs_node.name());
+  FBS_SET_STR_VAL(description_, fbs_node.doc_string());
+  FBS_SET_STR_VAL(domain_, fbs_node.domain());
+  since_version_ = fbs_node.since_version();
+  FBS_SET_STR_VAL(op_type_, fbs_node.op_type());  // index was set in ctor
+  node_type_ = static_cast<Node::Type>(fbs_node.type());
```

Comment:
comment could maybe be on a separate line before we set values from fbs_node, and may be slightly clearer to say 'index_ was set in ctor'